pkg/utils/utils.go

```diff
@@ -89,38 +104,49 @@ func InitInformerMap() {
 	}
 
+	// Create dynamic shared informer factory
+	DynamicKubeInformerFactory = dynamicinformer.NewDynamicSharedInformerFactory(DynamicKubeClient, time.Duration(rsyncTime)*time.Minute)
+
 	// Create shared informer factory
-	KubeInformerFactory = informers.NewSharedInformerFactory(KubeClient, time.Duration(rsyncTime)*time.Minute)
+	//KubeInformerFactory = informers.NewSharedInformerFactory(KubeClient, time.Duration(rsyncTime)*time.Minute)
 
 	// Init maps
 	ResourceInformerMap = make(map[string]cache.SharedIndexInformer)
 	AllowedEventKindsMap = make(map[EventKind]bool)
 	AllowedUpdateEventsMap = make(map[KindNS]config.UpdateSetting)
 
+	// for i, v := range botkubeConf.Resources {
+	// 	gvr := parseResourceArg(v.Name)
+	// 	fmt.Printf(("Will create Informer for: %v, GroupVersionResource: %v"), v.Name, gvr)
+	// 	ResourceInformerMap[strings.TrimSuffix(gvr.Resource, "s")] = DynamicKubeInformerFactory.ForResource(gvr).Informer()
+	// 	botkubeConf.Resources[i].Name = strings.TrimSuffix(gvr.Resource, "s")
+
+	// }
+
 	// Informer map
-	ResourceInformerMap["pod"] = KubeInformerFactory.Core().V1().Pods().Informer()
-	ResourceInformerMap["node"] = KubeInformerFactory.Core().V1().Nodes().Informer()
-	ResourceInformerMap["service"] = KubeInformerFactory.Core().V1().Services().Informer()
-	ResourceInformerMap["namespace"] = KubeInformerFactory.Core().V1().Namespaces().Informer()
-	ResourceInformerMap["replicationcontroller"] = KubeInformerFactory.Core().V1().ReplicationControllers().Informer()
-	ResourceInformerMap["persistentvolume"] = KubeInformerFactory.Core().V1().PersistentVolumes().Informer()
-	ResourceInformerMap["persistentvolumeClaim"] = KubeInformerFactory.Core().V1().PersistentVolumeClaims().Informer()
-	ResourceInformerMap["secret"] = KubeInformerFactory.Core().V1().Secrets().Informer()
-	ResourceInformerMap["configmap"] = KubeInformerFactory.Core().V1().ConfigMaps().Informer()
-
-	ResourceInformerMap["deployment"] = KubeInformerFactory.Apps().V1().Deployments().Informer()
-	ResourceInformerMap["daemonset"] = KubeInformerFactory.Apps().V1().DaemonSets().Informer()
-	ResourceInformerMap["replicaset"] = KubeInformerFactory.Apps().V1().ReplicaSets().Informer()
-	ResourceInformerMap["statefulset"] = KubeInformerFactory.Apps().V1().StatefulSets().Informer()
-
-	ResourceInformerMap["ingress"] = KubeInformerFactory.Networking().V1beta1().Ingresses().Informer()
-
-	ResourceInformerMap["job"] = KubeInformerFactory.Batch().V1().Jobs().Informer()
-
-	ResourceInformerMap["role"] = KubeInformerFactory.Rbac().V1().Roles().Informer()
-	ResourceInformerMap["rolebinding"] = KubeInformerFactory.Rbac().V1().RoleBindings().Informer()
-	ResourceInformerMap["clusterrole"] = KubeInformerFactory.Rbac().V1().ClusterRoles().Informer()
-	ResourceInformerMap["clusterrolebinding"] = KubeInformerFactory.Rbac().V1().RoleBindings().Informer()
+	ResourceInformerMap["pod"] = DynamicKubeInformerFactory.ForResource(parseResourceArg("pods.v1")).Informer()
```

Comment:
We can get rid of `ResourceInformerMap`. We can just iterate over resource config and register informers in controller.go